dotCMS/src/main/java/com/dotcms/metrics/RequestTracker.java

```diff
@@ -0,0 +1,278 @@
+        } catch (Exception e) {
+            Logger.warn(this, "Failed to track request end: " + e.getMessage(), e);
+        }
+    }
+    
+    /**
+     * Increment status code counter for metrics collection.
+     * 
+     * @param statusCode the HTTP status code
+     */
+    public void incrementStatusCode(int statusCode) {
+        String range = getStatusCodeRange(statusCode);
+        statusCodeCounts.computeIfAbsent(range, k -> new AtomicLong(0)).incrementAndGet();
+    }
+    
+    /**
+     * Increment endpoint counter for metrics collection.
+     * 
+     * @param uri the request URI
+     */
+    public void incrementEndpoint(String uri) {
+        String pattern = getEndpointPattern(uri);
+        if (pattern != null) {
+            endpointCounts.computeIfAbsent(pattern, k -> new AtomicLong(0)).incrementAndGet();
+        }
+    }
+    
+    // ====================================================================
+    // GETTER METHODS FOR METRICS COLLECTION
+    // ====================================================================
+    
+    /**
+     * Get the current number of active requests.
+     * 
+     * @return active request count
+     */
+    public long getActiveRequests() {
+        return activeRequests.get();
+    }
+    
+    /**
+     * Get the total number of requests processed.
+     * 
+     * @return total request count
+     */
+    public long getTotalRequests() {
+        return totalRequests.get();
+    }
+    
+    /**
+     * Get the total duration of all requests in milliseconds.
+     * 
+     * @return total duration in milliseconds
+     */
+    public long getTotalDuration() {
+        return totalDuration.get();
+    }
+    
+    /**
+     * Get the average request duration in milliseconds.
+     * 
+     * @return average duration in milliseconds
+     */
+    public double getAverageDuration() {
+        long total = totalRequests.get();
+        return total > 0 ? (double) totalDuration.get() / total : 0.0;
+    }
+    
+    /**
+     * Get count for specific status code range.
+     * 
+     * @param statusRange the status code range (e.g., "2xx", "4xx")
+     * @return count of responses in that range
+     */
+    public long getStatusCodeCount(String statusRange) {
+        AtomicLong count = statusCodeCounts.get(statusRange);
+        return count != null ? count.get() : 0;
+    }
+    
+    /**
+     * Get count for specific endpoint pattern.
+     * 
+     * @param endpointPattern the endpoint pattern (e.g., "/api/")
+     * @return count of requests to that endpoint pattern
+     */
+    public long getEndpointCount(String endpointPattern) {
+        AtomicLong count = endpointCounts.get(endpointPattern);
+        return count != null ? count.get() : 0;
+    }
+    
+    /**
+     * Calculate overall error rate percentage.
+     * 
+     * @return error rate as percentage (0-100)
+     */
+    public double getErrorRate() {
+        long total2xx = getStatusCodeCount("2xx");
```

Comment:
It would be nice to have an enum about this ResponseCode Ranges or so

Reply:
Updated to add StatusCodeRange enum